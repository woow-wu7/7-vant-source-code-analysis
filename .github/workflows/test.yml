name: CI

on: [push, pull_request]

jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      uses: pnpm/action-setup@v2.0.1
      with:
        version: 6.20.3

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run linter
      run: npm run lint

  test:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      uses: pnpm/action-setup@v2.0.1
      with:
        version: 6.20.3

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run test cases
      run: npm test

    - name: Upload coverage to Codecov  
      uses: codecov/codecov-action@v1
      with:
        token: ${{ secrets.CODECOV_TOKEN }}

  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      uses: pnpm/action-setup@v2.0.1
      with:
        version: 6.20.3

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Build
      run: npm run build
