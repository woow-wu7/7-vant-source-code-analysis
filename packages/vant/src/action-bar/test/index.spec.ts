import { ActionBar } from '..';
import { mount } from '../../../test';

test('should allow to disable safe-area-inset-bottom prop', () => {
  const wrapper = mount(ActionBar, {
    props: {
      safeAreaInsetBottom: false,
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});
