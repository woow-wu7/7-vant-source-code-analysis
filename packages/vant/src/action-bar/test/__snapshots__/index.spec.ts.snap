// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should allow to disable safe-area-inset-bottom prop 1`] = `
<div class="van-action-bar">
</div>
`;
